src/dbnode/persist/fs/commitlog/commit_log_test.go

```diff
@@ -354,23 +360,75 @@ func TestCommitLogWrite(t *testing.T) {
+			},
+		},
+		{
+			"Buffer almost full after first write. Second write almost fills the buffer",
+			[]testWrite{
+				{testSeries(0, "foo.bar", ident.NewTags(ident.StringTag("name1", "val1")), 127), time.Now(), 123.456, xtime.Second, randomByteArray(opts.FlushSize() - 200), nil},
+				{testSeries(1, "foo.baz", ident.NewTags(ident.StringTag("name2", "val2")), 150), time.Now(), 456.789, xtime.Second, randomByteArray(40), nil},
+			},
+		},
+		{
+			"Buffer almost full after first write. Second write almost fills 2*buffer total",
+			[]testWrite{
+				{testSeries(0, "foo.bar", ident.NewTags(ident.StringTag("name1", "val1")), 127), time.Now(), 123.456, xtime.Second, randomByteArray(opts.FlushSize() - 200), nil},
+				{testSeries(1, "foo.baz", ident.NewTags(ident.StringTag("name2", "val2")), 150), time.Now(), 456.789, xtime.Second, randomByteArray(40 + opts.FlushSize()), nil},
+			},
+		},
+		{
+			"Buffer almost full after first write. Second write almost fills 3*buffer total",
+			[]testWrite{
+				{testSeries(0, "foo.bar", ident.NewTags(ident.StringTag("name1", "val1")), 127), time.Now(), 123.456, xtime.Second, randomByteArray(opts.FlushSize() - 200), nil},
+				{testSeries(1, "foo.baz", ident.NewTags(ident.StringTag("name2", "val2")), 150), time.Now(), 456.789, xtime.Second, randomByteArray(40 + 2 * opts.FlushSize()), nil},
+			},
+		},
+		{
+			"Attempts to perform a write equal to the flush size",
+			[]testWrite{
+				{testSeries(0, "foo.bar", ident.NewTags(ident.StringTag("name1", "val1")), 127), time.Now(), 123.456, xtime.Second, randomByteArray(opts.FlushSize()), nil},
+			},
+		},
+		{
+			"Attempts to perform a write double the flush size",
+			[]testWrite{
+				{testSeries(0, "foo.bar", ident.NewTags(ident.StringTag("name1", "val1")), 127), time.Now(), 123.456, xtime.Second, randomByteArray(2 * opts.FlushSize()), nil},
+			},
+		},
+		{
+			"Attempts to perform a write three times the flush size",
+			[]testWrite{
+				{testSeries(0, "foo.bar", ident.NewTags(ident.StringTag("name1", "val1")), 127), time.Now(), 123.456, xtime.Second, randomByteArray(3 * opts.FlushSize()), nil},
+			},
+		},
 	}
 
-	// Call write sync
-	writeCommitLogs(t, scope, commitLog, writes).Wait()
+	for _, testCase := range testCases {
+		t.Run(testCase.testName, func(t *testing.T) {
+			defer cleanup(t, opts)
 
-	// Close the commit log and consequently flush
-	require.NoError(t, commitLog.Close())
+			commitLog := newTestCommitLog(t, opts)
 
-	// Assert writes occurred by reading the commit log
-	assertCommitLogWritesByIterating(t, commitLog, writes)
+			// Call write sync
+			writeCommitLogs(t, scope, commitLog, testCase.writes).Wait()
+
+			// Close the commit log and consequently flush
+			require.NoError(t, commitLog.Close())
+
+			// Assert writes occurred by reading the commit log
+			assertCommitLogWritesByIterating(t, commitLog, testCase.writes)
```

Comment:
At a glance, doesn't look like this assert checks that the returned data is the same as what we put in--maybe worth adding that check?

Reply:
assertCommitLogWritesByIterating performs a full check by reading log entries using commit log reader and matching them against log entries in our test case.

Reply:
Sg, I think I just missed it.